airflow/utils/db.py

```diff
@@ -1446,7 +1446,7 @@ class BadReferenceConfig:
 
         dangling_table_name = _format_airflow_moved_table_name(source_table.name, change_version, "dangling")
         if dangling_table_name in existing_table_names:
-            invalid_row_count = bad_rows_query.count()
+            invalid_row_count = get_query_count(bad_rows_query, session=session)
```

Comment:
Can we add a test to protect future changes here?

Reply:
Sure, I will add it

Reply:
@ephraimbuddy I've finally found the time to add the test, although it required more work than expected :)